sdk/core/src/main/java/com/klaviyo/coresdk/Klaviyo.kt

```diff
@@ -9,54 +10,126 @@ import com.klaviyo.coresdk.networking.requests.KlaviyoRequest
+     *
+     * @param email Email address for active user
+     */
+    fun setEmail(email: String) = apply {
         UserInfo.email = email
     }
 
     /**
-     * Creates a track request for the Klaviyo APIs and begins processing.
-     * Track requests track the triggering of a specified [KlaviyoEvent] in Klaviyo
+     * Assigns a phone number to the current UserInfo
      *
-     * @param customerProperties A map of customer property information.
-     * Defines the customer that triggered this event
-     * @param properties A map of event property information.
-     * Additional properties associated to the event that are not for identifying the customer
+     * UserInfo is saved to keep track of current profile details and
+     * used to autocomplete analytics requests with profile identifier when not specified
+     *
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
+     *
+     * @param phone Phone number for active user
      */
-    fun track(
-        event: KlaviyoEvent,
-        customerProperties: KlaviyoCustomerProperties,
-        properties: KlaviyoEventProperties? = null
-    ) {
-        val request = TrackRequest(KlaviyoConfig.apiKey, event, customerProperties, properties)
-        processRequest(request)
+    fun setPhone(phone: String) = apply {
+        UserInfo.phone = phone
     }
 
     /**
-     * Creates an identify request for the Klaviyo APIs and begins processing.
+     * Clears all stored UserInfo identifiers (e.g. email or phone)
+     *
+     * This should be called whenever an active user in your app is removed
+     * (e.g. after a logout)
+     */
+    fun reset() = apply {
+        UserInfo.reset()
+    }
+
+    //endregion
+
+    //region Analytics API
+
+    /**
+     * Queues a request to identify profile properties to the Klaviyo API
      * Identify requests track specific properties about a user without triggering an event
      *
      * @param properties A map of properties that define the user
      */
-    fun identify(properties: KlaviyoCustomerProperties) {
+    fun createProfile(properties: KlaviyoCustomerProperties) {
         val request = IdentifyRequest(KlaviyoConfig.apiKey, properties)
         processRequest(request)
     }
 
+    /**
+     * Queues a request to track a [KlaviyoEvent] to the Klaviyo API
+     * The event will be associated with the profile specified by the [KlaviyoCustomerProperties]
+     * If customer properties are not set, this will fallback on the current profile identifiers
+     *
+     * @param customerProperties A map of customer property information.
+     * Defines the customer that triggered this event
+     * @param properties A map of event property information.
+     * Additional properties associated to the event that are not for identifying the customer
+     */
+    fun createEvent(
+        event: KlaviyoEvent,
+        customerProperties: KlaviyoCustomerProperties? = null,
```

Comment:
nit: might move event first...

Reply:
Had the same thought while I was writing this readme section!